src/Lean/Environment.lean

```diff
@@ -117,6 +117,9 @@ declared by users are stored in an environment extension. Users can declare new
 using meta-programming.
 -/
 structure Environment where
+  /-- The constructor of `Environment` is private to protect against modification
+  that bypasses the kernel. -/
```

Comment:
I think it would be better to say "discourage" or "inhibit" rather than "protect against", given the existence of `open private`! 

(I don't want the comment to encourage someone feeling clever to announce that `open private` leads to unsoundness.)

Reply:
another possible word is "defend against", I don't think this implies indefeasibility. But I didn't really want to go into an extended discussion about the soundness implications of violating privacy here.